src/logic/Windows.swift

```diff
@@ -162,8 +162,8 @@ class Windows {
     static func detectTabbedWindows() {
         let cgsWindowIds = Spaces.windowsInSpaces(Spaces.idsAndIndexes.map { $0.0 }, [])
         list.forEach {
-            if let cgWindowId = $0.cgWindowId {
-                $0.isTabbed = !$0.isMinimized && !$0.isHidden && !cgsWindowIds.contains(cgWindowId)
+            if let cgWindowId = $0.cgWindowId, !$0.isMinimized && !$0.isHidden {
```

Comment:
I think one scenario that this would struggle with is when a window with tabs has been minimized/hidden before AltTab runs. AltTab would then shows those tabs as windows since their `isTabbed` would be at default value `false`

Reply:
The bigger flaw is that it does not take into account the creation/deletion of a group of tabs and its minimizing/hiding between two invocations of AltTab.

Reply:
I've experienced with detecting tabs better.

I found that there is no need to check for minimized/hidden. If we use `CGSCopyWindowsWithOptionsAndTags` with the `.minimizedAndTabbed` option (and actually use the option in the method).

It seems to handle every case I tested.

Except one of course. When you make 2 windows of 2 tabs each, then use the menubar action `Window > Merge All Windows`. Then, even though there is now only 1 window with 4 tabs, `CGSCopyWindowsWithOptionsAndTags` still returns an array containing the 2 tabs that were the visible tabs of the previous 2 windows. Even though now there is only 1 active tab in the 4-tabs window, the API still returns 2 tabs.

I experimented with using `axUiElement.children()`, and checking if there is a child with `role() == kAXTabGroupRole`. First of all, using AX calls is really bad when the user invokes AltTab. We want to avoid it, as these calls can resolve **seconds** later sometimes. That's why we normally process them on a background thread. But even if I ignore this technical issue, it's still not enough because it doesn't let us distinguish a window with no tabs (has no tabs-bar), from a non-visible tab (has no tabs-bar).

I also re-read, and advise you do the same, this ticket were I explained many approaches I tried in the past: https://github.com/lwouis/alt-tab-macos/issues/1540

My conclusion for now is that there is no perfect way still. All approaches have issues.

I think I'll release a variation of this commit, with the `minimized`/`hidden` checks removed, and instead adding `.minimizedAndTabbed` to `CGSCopyWindowsWithOptionsAndTags`. That seems to me like the most robust/comprehensive check we can do for the time being.

Reply:
I had not caught this exception. It can be partially mitigated by using the option only for minimized or hidden windows.